python/tskit/trees.py

```diff
@@ -5442,3 +5442,39 @@ def newick_trees(self, precision=3, breakpoints=None, Ne=1):
             "This method is no longer supported. Please use the Tree.newick"
             " method instead"
         )
+
+    def nexus(self, precision=14):
+        """
+        Returns a nexus encoding <https://en.wikipedia.org/wiki/Nexus_file>`_
+        of this tree. Trees along the sequence are listed sequentially in
+        the TREES block. The tree spanning the interval :math:`[x, y)`` is
+        given the name "tree_x_y". Spatial positions are written at the
+        specified precision.
+
+        :param int precision: The numerical precision with which branch lengths
+            and tree positions are printed.
+        :param dict node_labels: If specified, show custom labels for the nodes
+            that are present in the map. Any nodes not specified in the map will
+            not have a node label.
+        :return: A nexusu representation of this tree.
+        :rtype: str
+        """
+        leaf_labels = {u: str(u + 1) for u in self.samples()}
```

Comment:
The nexus format contains the nexus outputs in the TREE block, which use these labels by default.
Instead, I could use the default label itself and use the translate block to convert the node ids to the ones used by newick. The code could then be used for custom translations too.

Reply:
Ah, I see, and then 1 to n is used to index into the TRANSLATE block? OK, then we want to make sure that the labels in the newick trees are 1 to n (u + 1 won't do it, as the samples are not necessarily 0 to n - 1), and use the TRANSLATE block to map these to their actual node IDs, as the default behaviour.

Reply:
There is a distinction between "leaves" and "samples" here too which we want to be careful with. Samples are special nodes we are interested in. Samples are not necessarily leaves, and leaves are not necessarily samples. https://tskit.readthedocs.io/en/latest/data-model.html#definitions

Reply:
In that case, I should follow the code in TestNewick class and use tree.leaves().
In which case are the leaves not labelled 0 to n-1? I might need to change the code accordingly

Reply:
We are more interested in samples than leaves, so it would be better to use samples. Have a look in test_topology for examples in which leaves are not 0 to n - 1, there's all sorts of weird topologies that can happen that are different to the basic msprime simualtion.